src/_guides/language/effective-dart/usage.md

```diff
@@ -441,33 +444,110 @@ var reciprocals = stuff.map((n) => 1 / n).cast<double>();
 {% endprettify %}
 
 
-### AVOID using `cast()` or `retype()`.
+### AVOID using `cast()`.
 
 This is the softer generalization of the previous rule. Sometimes there is no
 nearby operation you can use to fix the type of some object. Even then, when
-possible avoid using `cast()` or `retype()` to "change" a collection's type.
+possible avoid using `cast()` to "change" a collection's type.
 
 Prefer any of these options instead:
 
-*   Change the code where the collection is first created so that it has the
-    right type.
+#### Create it with the right type
```

Comment:
Although I appreciate the sample code, I miss seeing all the options in one windowful. How about keeping the list, but making each item a link to the subheading. Like this:

```
...instead:

* [Create it with the right type](#create-it-with-the-right-type)
* [Cast the elements on access](#cast-the-elements-on-access)
* ...

#### Create it with the right type
```

Reply:
Done. I liked seeing everything in a list too. I wish the site supported code snippets inside bullet lists, and I wish the code snippets were a little more compact. Maybe something to consider for the next redesign.

Reply:
Could you file an issue about that? If you could point to an existing example of formatting that you really like, that would be helpful.